### ISSUE TITLE
Add option to assign an FHRP group to a tunnel termination

### ISSUE THREAD

**Issue by dreng** [netbox-community/netbox#17943]
### NetBox version

v4.1.6

### Feature type

Change to existing functionality

### Triage priority

N/A

### Proposed functionality

In the "add a new tunnel" screen, please add an option "FHRP Group" to the first and second termination type. If selected, the following field (that switches between "Device" and "Virtual Machine") should change the Caption to "FHRP Group" and query FHRP Groups.

### Use case

With a redundant tunnel hub, it does not reflect the reality that the termination must be assigned to a device. Instead, the termination should be assigned to a virtual interface which does not belong to a specific device but to a FHRP Group.

### Database changes

Not sure

### External dependencies

None

**Comment by wlnx** [comment]
I'd like to point out that in fact a tunnel termination point is not an interface but an IP address (or even an IP address set).

**Comment by jeremystretch** [comment]
Yeah, I don't see how this is intended to work. Tunnels must terminate to interfaces.

**Comment by wlnx** [comment]
> Yeah, I don't see how this is intended to work. Tunnels must terminate to interfaces.

Well... I have two cases:
* One tunnel terminates to the IP assigned to VRRP (two interfaces belonging to different devices)
* Three different tunnels terminate to one interface that has three IP-addresses assigned.

When we talk about physical topologies, any cable can be plugged into a port. But tunnels are not physical entities, when we set them up, we point a tunnel client to an IP address (that may be a multicast, e. g.), thus, there is no tunnel-to-interface binding, in general.
Of course, keeping in mind the topology that has only one IP to one interface bound, tunnel may be considered to terminate to interface. But the approach is a HUGE limitation.

**Comment by dreng** [comment]
> Tunnels must terminate to interfaces.

I agree. But in an active/passive configuration, how would you suggest to document such networks? Let's say, wie have two routers in an active/passive HA cluster, R1 and R2. In the current situation, you have to define the tunnel termination to an interface of just one of this devices, let's say tun0 of R1. If R1 fails, R2 becomes active automatically, which means that the tunnel termination switches to tun0 of R2. This situation lets the documented state in NetBox instantly become incorrect. I currently don't see a good workaround without changing NetBox code. Suggestions are welcome.
